PLT-2784 Fixed JS errors, wrapped around nav shortcuts

DavidLu1997 opened the pull request:
ALT [up arrow]: Previous channel/DM in the left sidebar list
ALT [down arrow]: Next channel/DM in the left sidebar list
ALT SHIFT [up arrow]: Previous channel/DM with unread messages
ALT SHIFT [down arrow]: Next channel/DM with unread messages

All shortcuts now wrap around.

@jwilander please review

enahum commented:
+1

jwilander commented on webapp/components/navbar.jsx:
```diff
@@ -187,18 +195,29 @@ export default class Navbar extends React.Component {
-            if (nextIndex !== curIndex && ChannelStore.getUnreadCount(allChannels[nextIndex].id).msgs !== 0 || ChannelStore.getUnreadCount(allChannels[nextIndex].id).mentions !== 0) {
-                nextChannel = allChannels[nextIndex];
-                GlobalActions.emitChannelClickEvent(nextChannel);
+            if (nextIndex >= 0 && nextIndex < allChannels.length) {
```
Isn't this check a bit redundant? Won't `nextIndex` always be `>= 0` and `< allChannels.length` at this point due to the conditions you added above?

crspeller commented:
This whole wrapping thing creates quite confusing code. Can we break some if it into functions? Try to simplify? (my gut is screaming Modulo Operator) Maybe generalize it so it can be used in multiple places requiring wrapping on a value?

DavidLu1997 commented:
Simplified the wrapping stuff. I found out the hard way that JS modulus does not work on negatives. Removed some redundant checking code.

I can split it up into functions and put the wrapping function in Utils.

crspeller commented:
@DavidLu1997 Apparently this will work:
```javascript
function mod(n1, n2) {
    return ((n1 % n2) + n2) % n2;
}
```

https://stackoverflow.com/questions/18618136/how-to-calculate-modulo-of-negative-integers-in-javascript

DavidLu1997 commented:
Added mod to `utils.jsx`

jwilander commented:
+1, waiting on @crspeller to confirm he's happy with changes before merging

crspeller commented:
@DavidLu1997 I noticed you are grabbing values from the global store. We are trying to avoid this because there is no guarantee it is loaded yet. (you can see Joram's scratch doc for an explanation). Can we have your help moving store calls to using state that is checked for validity?

jwilander commented:
Will work with DL to resubmit
